ui/panels-plugin/src/plugins/time-series-chart/utils/palette-gen.ts

```diff
@@ -23,3 +27,26 @@ export function getRandomColor(identifier: string): string {
   const color = `hsla(${~~(180 * hash)},50%,50%,0.8)`;
   return color;
 }
+
+/**
+ * Get line color as well as color for tooltip and legend, account for whether palette is 'Cateogrical' or 'Auto' (generative)
+ */
+export function getSeriesColor(
+  name: string,
+  seriesCount: number,
+  palette: string[],
+  paletteKind: PaletteOptions['kind'] = 'Auto'
+): string {
+  if (paletteKind === 'Auto') {
+    return getRandomColor(name);
+  }
+
+  if (paletteKind === 'Categorical' && Array.isArray(palette)) {
```

Comment:
So the default palette is what is defined in the EChart theme [here](https://github.com/perses/perses/blob/v0.25.0/ui/components/src/utils/theme-gen.ts#L42), since that theme gets passed into our ECharts wrapper component, it automatically takes effect. If we used the ECharts legend and tooltip and it would just work there too, but we built those custom so had to loop through the palette explicitly to populate graphData.legendItems